maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -955,27 +957,20 @@ protected final Toolchain getToolchain()
 
         if ( jdkToolchain != null )
         {
-            // Maven 3.3.1 has plugin execution scoped Toolchain Support
             try
             {
-                Method getToolchainsMethod =
-                    toolchainManager.getClass().getMethod( "getToolchains", MavenSession.class, String.class,
-                        Map.class );
-
-                @SuppressWarnings( "unchecked" )
+                Method getToolchainsMethod = ReflectionUtils.getMethod( getToolchainManager(), "getToolchains" );
                 List<Toolchain> tcs =
-                    (List<Toolchain>) getToolchainsMethod.invoke( toolchainManager, getSession(), "jdk",
-                        jdkToolchain );
-
+                    (List<Toolchain>) ReflectionUtils.invokeMethodWithArray( getToolchainManager(),
+                        getToolchainsMethod, getSession(), "jdk" );
                 if ( tcs != null && !tcs.isEmpty() )
                 {
                     tc = tcs.get( 0 );
                 }
             }
-            catch ( NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException
-                | InvocationTargetException e )
+            catch ( Exception x )
```

Comment:
Why the `try-catch` is used? The methods `invokeMethodWithArray` and `getMethod` don't throw any exception. Maven API is stable and backwards compatile. Thus if we are not able to call Maven 3.3.x API then it means we are on 3.0 - 3.2 and we go to the fallback. But this will never crash if we use right signatures.

Reply:
Try to use another pattern. Instead if `getMethod` use `tryGetMethod` which returns null for Maven 3.0 - 3.2. The method `invokeMethodWithArray` should be called if not null which means we are at Maven 3.3+.